e2etest/declarativeTestFiles.go

```diff
@@ -97,6 +113,51 @@ func (op objectProperties) sizeBytes(a asserter, defaultSize string) int {
 	return 0
 }
 
+func (op objectProperties) DeepCopy() objectProperties {
+	ret := objectProperties{}
+	ret.isFolder = op.isFolder
+
+	if op.size != nil {
+		val := *&op.size
+		ret.size = val
+	}
+
+	if op.contentHeaders != nil {
+		ret.contentHeaders = op.contentHeaders.DeepCopy()
+	}
+	
+	ret.nameValueMetadata = make(map[string]string)
+	for k, v := range op.nameValueMetadata {
+		ret.nameValueMetadata[k] = v
+	}
+
+	ret.blobTags = make(map[string]string)
+	for k, v := range op.blobTags {
+		ret.blobTags[k] = v
+	}
+
+	if op.creationTime != nil {
+		time := *op.creationTime
+		ret.creationTime = &time
+	}
+
+	if op.lastWriteTime != nil {
+		time := *op.lastWriteTime
+		ret.lastWriteTime = &time
+	}
+
+	if op.smbAttributes != nil {
+		val := *op.smbAttributes
+		ret.smbAttributes = &val
+	}
+
+	if op.smbPermissionsSddl != nil {
+		ret.smbPermissionsSddl = op.smbPermissionsSddl
+	}
+
+	return ret
+}
+
```

Comment:
Great! 
You really did the hard-work and copied every single attribute. I remember us discussing about using libraries to do that. Will that not work here?

Reply:
Couldnt find appropriate library. Had problems with unexported fields.